Formula/vineyard.rb

```diff
@@ -0,0 +1,48 @@
+class Vineyard < Formula
+  include Language::Python::Virtualenv
+
+  desc "In-memory immutable data manager. (Project under CNCF)"
+  homepage "https://v6d.io"
+  url "https://github.com/v6d-io/v6d/releases/download/v0.5.4/v6d-0.5.4.tar.gz"
+  sha256 "ac1d68a7e61778cca737b139fc60a5e05de4f3707b61937ca31a7f9b2c31d21b"
+  license "Apache-2.0"
+
+  depends_on "cmake" => :build
+
+  depends_on "apache-arrow"
+  depends_on "boost"
+  depends_on "etcd-cpp-apiv3"
+  depends_on "gflags"
+  depends_on "glog"
+  depends_on "libgrape-lite"
+  depends_on "open-mpi"
+  depends_on "openssl@1.1"
+  depends_on "python@3.9"
+
+  fails_with gcc: "5"
+
+  def install
+    virtualenv_create(buildpath/"venv", "python3")
+
+    # libclang is a python package that required during the build stage,
+    # and only available as prebuilt wheels and cannot be installed by
+    # venv.pip_install (which implies --no-binary flag).
+    system buildpath/"venv"/"bin/pip", "install", "libclang"
+
+    system "cmake", "-S", ".", "-B", "build",
+                    "-DCMAKE_CXX_STANDARD=14",
+                    "-DCMAKE_CXX_STANDARD_REQUIRED=TRUE",
+                    "-DPYTHON_EXECUTABLE=#{buildpath}/venv/bin/python",
+                    "-DUSE_EXTERNAL_ETCD_LIBS=ON",
+                    "-DUSE_EXTERNAL_TBB_LIBS=ON",
+                    "-DBUILD_VINEYARD_TESTS=OFF",
+                    "-DOPENSSL_ROOT_DIR=#{Formula["openssl@1.1"].opt_prefix}",
+                    *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    assert_match(/^vineyardd version.*/, shell_output("#{Formula["vineyard"].bin}/vineyardd --version"))
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.